test/fixtures/nested_block_with_array_do_end.rb

```diff
@@ -0,0 +1,9 @@
+groups.each do |group|
```

Comment:
Let's add some more complicated tests. The key to getting this right will be ensuring that nested arrays and hashes are being properly handled.